### ISSUE TITLE
Check dtype arguments for Function types

### ISSUE THREAD

**Issue by moble** [numba/numba#4315]
## Feature request

As mentioned [here](https://github.com/numba/numba/issues/3993#issuecomment-484851921), numba chokes on creating arrays with statements like `np.zeros((2, 2), dtype=int)` or `np.zeros((2, 2), dtype=np.int)` [and similarly for `float` and `complex`], even though these are among the most common ways to construct arrays.  It turns out the correct way to do this would be with `np.zeros((2, 2), dtype=np.int_)`, which is something that — even having extensive experience with numpy and its C API — I had not thought to try until finding the solution in that other thread.  And in any case, it certainly seems to work against the goal of numba as being as simple as decorating normal python code.

@stuartarchibald suggested that the solution would be
> something that especially checks the dtype and if the type is identified as a `Function` class type it hints that you probably meant an equivalent concrete NumPy type

and noted that this is "not quite the same sort of code path as #3942".

**Comment by seibert** [comment]
Yes, this request makes sense.  We should be able to map Python types to dtypes with the same convention that NumPy uses.

**Comment by jstac** [comment]
Thanks @seibert.  It would be great to have this feature.  My colleagues and I are trying to convince all those MATLAB-loving economists make the switch and [coordinate on Numba](https://lectures.quantecon.org/py/numba.html) as a way to write fast code.  It's very helpful if simple NumPy functions to create arrays are all processed smoothly.

CC @shizejin
